lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
I was going by [this bit of the documentation](https://dev.mysql.com/doc/refman/5.6/en/set-transaction.html), but perhaps I misinterpreted it:

> #### Scope of Transaction Characteristics
>
> You can set transaction characteristics globally, for the current session, or for the next transaction:
>
> * With the GLOBAL keyword, the statement applies globally for all subsequent sessions. Existing sessions are unaffected.
>
> * With the SESSION keyword, the statement applies to all subsequent transactions performed within the current session.
>
> * Without any SESSION or GLOBAL keyword, the statement applies to the next (not started) transaction performed within the current session. Subsequent transactions revert to using the SESSION isolation level. 

Anyway, I'll make sure it gets reset at the end of the script, along with the move away from the `readUncommitted` approach that you describe in another comment.